src/include/catalog/table_handle.h

```diff
@@ -0,0 +1,212 @@
+#pragma once
+
+#include <iostream>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include "catalog/attribute_handle.h"
+#include "catalog/catalog.h"
+#include "catalog/catalog_defs.h"
+#include "catalog/catalog_sql_table.h"
+#include "loggers/catalog_logger.h"
+#include "storage/sql_table.h"
+#include "transaction/transaction_context.h"
+#include "type/transient_value.h"
+#include "type/transient_value_factory.h"
+
+namespace terrier::catalog {
+
+class Catalog;
+class AttributeHandle;
+
+/**
+ * A table entry represent a row in pg_tables catalog.
+ *
+ * A row in pg_tables needs information from pg_class, pg_namespace, pg_tablespace
+ *    1) it needs schemaname from pg_namespace
+ *    2) it needs tablename from pg_class
+ *    3) it needs tablespace from pg_tablespace
+ *
+ * The class uses a vector of size 3 to stores 3 rows in these three tables.
+ *    rows_[0] stores a row in pg_namespace which contains namespace information for this TableEntry
+ *    rows_[1] stores a row in pg_class which contains table information for this TableEntry
+ *    rows_[2] stores a row in pg_tablespace which contains table information for this TableEntry
+ */
+class TableEntry {
+ public:
+  /**
+   * Constructs a table entry in pg_tables.
+   * TODO(yeshengm): consider using universal reference here??? we are now forcing a rvalue ref.
+   *
+   * @param oid the table oid
+   * @param row a row in pg_class that represents this table
+   * @param txn the transaction which wants the entry
+   * @param pg_class a pointer to the pg_class catalog
+   * @param pg_namespace a pointer to pg_namespace
+   * @param pg_tablespace a pointer to tablespace
+   */
+  TableEntry(table_oid_t oid, std::vector<type::TransientValue> &&row, transaction::TransactionContext *txn,
+             std::shared_ptr<SqlTableRW> pg_class, std::shared_ptr<SqlTableRW> pg_namespace,
+             std::shared_ptr<SqlTableRW> pg_tablespace)
+      : oid_(oid),
+        txn_(txn),
+        pg_class_(std::move(pg_class)),
+        pg_namespace_(std::move(pg_namespace)),
+        pg_tablespace_(std::move(pg_tablespace)) {
+    rows_.resize(3);
+    rows_[1] = std::move(row);
+  }
+
+  /**
+   * Get the value for a given column number
+   * @param col_num the column number
+   * @return the value
+   */
+  const type::TransientValue &GetColInRow(uint32_t col_num) {
```

Comment:
This method should be private. We don't want to expose any of the internals of how we are implementing the view.